src/promptflow/promptflow/_sdk/_service/apis/run.py

```diff
@@ -71,15 +72,30 @@ def post(self):
             run_file = Path(temp_dir) / "batch_run.yaml"
             with open(run_file, "w", encoding="utf-8") as f:
                 dump_yaml(run_dict, f)
-            cmd = f"pf run create --file {run_file}"
+            cmd = [
+                "pf",
+                "run",
+                "create",
+                "--file",
+                str(run_file),
+                "--user-agent",
+                build_pfs_user_agent(),
+            ]
+
             if sys.executable.endswith("pfcli.exe"):
-                cmd = f"pfcli {cmd}"
+                cmd = ["pfcli"] + cmd
             process = subprocess.Popen(cmd, stdout=subprocess.PIPE, stderr=subprocess.STDOUT, shell=True)
             stdout, _ = process.communicate()
             if process.returncode == 0:
-                run_op = RunOperations()
-                run = run_op.get(name=run_name)
-                return jsonify(run._to_dict())
+                try:
+                    run = get_client_from_request().runs._get(name=run_name)
+                    return jsonify(run._to_dict())
+                except RunNotFoundError as e:
+                    raise RunNotFoundError(
+                        f"Failed to get the submitted run: {e}\n"
+                        f"Used command: {' '.join(shlex.quote(arg) for arg in cmd)}\n"
```

Comment:
does this work properly on different platforms?  I searched the shlex document and found this line.
Warning The shlex module is only designed for Unix shells.

Reply:
this is a string to string convert. it will return a unix style shell command but:
1. may work for Windows & Mac in most cases too, given I just want to make sure that spaces are quoted
2. won't bring extra exceptions as this is a string convert and we won't execute the generated command

It can be misleading in some corner cases when unix style command is different from windows/mac style command, but it won't bring effect as this is an internal API and this error message is mainly for debugging